File: 135022193-main/2024/Done/sentimental-readability/readability.py
from cs50 import get_string

letters = 0
words = 1
sentences = 0


def main():
    text = get_string("Text: ")

    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    # L is average number of letters per 100 words
    L = letters / words * 100

    # S is average number of senteces per 100 words
    S = sentences / words * 100

    grade = 0.0588 * L - 0.296 * S - 15.8
    grade = round(grade)

    print_grade(grade)


def count_letters(text):
    sumL = 0

    for i in range(len(text)):
        if (text[i].isalpha() == True):
            sumL += 1
    return sumL


def count_words(text):
    sumW = 1

    for i in range(len(text)):
        if (text[i] == " "):
            sumW += 1
    return sumW


def count_sentences(text):
    sumS = 0

    for i in range(len(text)):
        if (text[i] == "." or text[i] == "!" or text[i] == "?"):
            sumS += 1
    return sumS


def print_grade(grade):
    if grade >= 16:
        print("Grade 16+")
    elif grade < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {grade}")


main()

File: 135022193-main/2023/CS50 Done/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

float count_letters(string text);
float count_words(string text);
float count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");
    float letters = count_letters(text);
    float words = count_words(text);
    float sentences = count_sentences(text);

    float L = letters / words * 100;
    float S = sentences / words * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;

    if(index>=1 && index<=16)
    {
        printf("Grade %g\n", round(index));
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Before Grade 1\n");
    }
}


float count_letters(string text)
{
    float letters = 0;

    for(int i = 0, len = strlen(text); i < len; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    return letters;
}


float count_words(string text)
{
    float words = 0;

    for(int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == ' ')
        {
            words++;
        }
    }
    return words + 1;

}


float count_sentences(string text)
{
    float sentences = 0;

    for(int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }

    }
    return sentences;
}